### PR TITLE
Add grammar specification

### LINKED ISSUE #46: Grammar in Repo rather than Wiki
We should probably put the specification for the grammar, and any related stuff (like recommendations or common uses), in the repo itself instead of the wiki. This way, we can modify and discuss the spec in pull requests; and the spec will be protected from a random person deciding to hit the "edit" button.

### PR DESCRIPTION
**Pull request by torkleyy**
Fixes #46

### CODE REVIEW

File: docs/grammar.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+## Commas
+
+```ebnf
+comma = ws, ",", ws
+```
+
+## Extensions
+
+```ebnf
+extensions = { "#", ws, "!", ws, "[", ws, extensions_inner, ws, "]", ws }
```
Comment (kvark) [inline docs/grammar.md]: I don't see any reason to support `ws` between `#` and `!`

File: docs/grammar.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+extensions_inner = "enable", ws, "(", extension_name, { comma, extension_name }, [comma], ws, ")" 
+```
+
+For the extension names see the [`extensions.md`][exts] document.
+
+## Value
+
+```ebnf
+value = unsigned | signed | float | string | char | bool | option | list | map | tuple | struct | enum_variant
```
Comment (kvark) [inline docs/grammar.md]: I don't know much about grammar specs, but can we have something that is more formal? E.g. something like `ron.ebnf` (or any other notation) that is parse-able and validate-able.

File: docs/grammar.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+```
+
+## Numbers
+
+```ebnf
+digit = "0" | "1" | "2" | "3" | "4" | "5" | "6" | "7" | "8" | "9"
+unsigned = digit, { digit }
```
Comment (kvark) [inline docs/grammar.md]: we've got to support other numerical systems (I know we don't support them atm)

File: docs/grammar.md (hunk 1 above)
Comment (torkleyy) [reply docs/grammar.md]: That's what Rust allows, too. And that's what I implemented. Should I change the implementation?

File: docs/grammar.md (hunk 2 above)
Comment (torkleyy) [reply docs/grammar.md]: I've thought about putting everything into one file and creating automated tests for our parser with it. But I don't have that much time and I think this is useful as is.

File: docs/grammar.md (hunk 3 above)
Comment (torkleyy) [reply docs/grammar.md]: Agreed

**Pull request merged**